src/RepositoryProxy.php

```diff
@@ -43,6 +45,8 @@ public function count(): int
 
     public function getIterator(): \Traversable
     {
+        // TODO: $this->repository is set to ObjectRepository, which is not
+        //       iterable. Can this every be another RepositoryProxy?
         if (\is_iterable($this->repository)) {
             return yield from $this->repository;
         }
```

Comment:
Psalm says "`$this->repository` is never iterable", because it's typehinted as `ObjectRepository`. Is this indeed a redundant if statement, or are there situations where `$this->repository` is already a `RepositoryProxy`? (which case we need to update its PHPdoc to `ObjectRepository<TProxiedObject>|RepositoryProxy<TProxiedObject>`)

Reply:
There isn't a situation where this `$this->repository` could already be a `RepositoryProxy` but I think it's valid that the `ObjectRepository` could be iterable. In my apps, all my repository's are instances of `ObjectRepository`, iterable and countable.

Reply:
Could we update the PHPdoc to `ObjectRepository<TProxiedObject>|iterable<TProxiedObject>|\Countable`?

Reply:
I believe the [union type](https://psalm.dev/docs/annotating_code/type_syntax/union_types/) is considered exclusive. So doing `ObjectRepository|iterable|\Countable` means that this property is either `ObjectRepository` or `iterable` or `\Countable`. This means that all code using `$this->repository` as if it's always `ObjectRepository` will be considered by Psalm.

Psalm uses the [intersection type (e.g. `ObjectRepository&iterable`)](https://psalm.dev/docs/annotating_code/type_syntax/intersection_types/) as an inclusive type. However this means that any code setting the property to an instance of `ObjectRepository` that isn't `iterable` will produce an error.

However, in this case the issue is ignored by the current Psalm level. So let's just ignore this for now. I think the best solution - if this becomes a real error - is to add `@psalm-suppress ...` (this is like saying "I know this strictly is wrong, but it's here for a reason").